ui/admin/app/routes/scopes/scope/roles/role/index.js

```diff
@@ -22,7 +22,9 @@ export default class ScopesScopeRolesRoleIndexRoute extends Route {
       : (await this.store.query('scope', { scope_id: currentScope.id })).map(
           (scope) => ({
             model: scope,
-            subScopes: this.store.query('scope', { scope_id: scope.id }),
+            subScopes: this.store
+              .query('scope', { scope_id: scope.id })
+              .then((models) => models.map((model) => model)),
```

Comment:
What's the purpose of this map?

Do we need to `.then` this, can't we await the query?

Reply:
good point. Actually, `.map` isn't needed here! I'll remove that. 

since the query returns an array, using await like the below code block, was just returning an unresolved array of promises. I could try to use `promise.all`, but using `.then` seemed simpler. 

>    const subScopes = currentScope.isProject
>       ? []
>       : (await this.store.query('scope', { scope_id: currentScope.id })).map(
>           async (scope) => ({
>             model: scope,
>             subScopes: await this.store.query('scope', { scope_id: scope.id }),
>           })
>         );

Reply:
Oh interesting 🤔 , it doesn't return a promise that resolves an array, it's an array of promises? That seems surprising since it seems [docs](https://api.emberjs.com/ember-data/4.12/classes/Store/methods/query?anchor=query) say `store.query` just returns a promise